src/Raven.Server/Documents/TransactionOperationsMerger.cs

```diff
@@ -1079,12 +1080,21 @@ private void NotifyOnThreadPool(List<MergedTransactionCommand> cmds)
             TaskExecutor.Execute(DoCommandsNotification, cmds);
         }
 
-        private void RunEachOperationIndependently(List<MergedTransactionCommand> pendingOps)
+        private void RunEachOperationIndependently(List<MergedTransactionCommand> pendingOps, Exception exception)
         {
             try
             {
-                foreach (var op in pendingOps)
+                int lastIndex = pendingOps.Count - 1;
+                for (int i = 0; i < pendingOps.Count; i++)
                 {
+                    if (i == lastIndex && exception is ConcurrencyException)
+                    {
+                        pendingOps[i].Exception = exception;
+                        continue;
+                    }
+
```

Comment:
I don't understand this?

Reply:
I guess it is an optimization, to say that if we already failed on `ConcurrencyException` no need to re-run, because we assume we will fail again.

Reply:
Ok, so the optimization is not worth the special treatment.
I will revert that commit